src/vsr/replica.zig

```diff
@@ -5336,10 +5336,12 @@ pub fn ReplicaType(
 
             self.sync_reclaim_tables();
 
-            // Request outstanding committed headers to advance our op number:
+            // Request outstanding possibly committed headers to advance our op number:
             // This handles the case of an idle cluster, where a backup will not otherwise advance.
             // This is not required for correctness, but for durability.
-            if (self.op < self.op_repair_max()) {
+            if (self.op < self.op_repair_max() or
+                (self.status == .normal and self.op < self.view_headers.array.get(0).op))
```

Comment:
`self.status == .normal and` is needed because, during view change, primary truncates its op&log before it updates its view headers.

Reply:
To check my understanding here:

- The prepares coming in while we are in normal mode include a view number, but that is the view that they were _originally_ prepared in, which might be behind the cluster's current view.
- Since they are behind the current view (but ahead of our `self.op`) the lagging (normal) replica (which is in the same view as the primary) has no way of knowing that they are indeed part of its current view, so it drops them.
- When we receive an SV in normal mode (for the same view that we are currently in) it is allowed to repair our headers and advance our op.

Is it possible for the lagging replica to get stuck at this step though? That is, it wants to request a SV to advance `self.op`, but if it is still in the old checkpoint (for example, because it needs to `repair_prepare()` some corruption before it can commit), then it can't actually install any headers from the SV yet. But we still "block" `repair()` from calling `repair_prepare()` in that case. Should their be an additional `and self.op_checkpoint_trigger() < self.op` clause?

Reply:
>Is it possible for the lagging replica to get stuck at this step though?

Yes! And that's exactly the issue I hit when I implemented the fix initially. In this second attempt, I am also moving the `return` _out_ of this if, so that we return _only_ if we send RSV due to the first condition of the branch. That is, for the new case we specifically avoid blocking the repair. 

To be fair, I am not quite happy with the conditions here --- they look somewhat adhoc, and it might be hard to reverse-engineer the logic behind them without `git blame`

Reply:
Yeah, I actually started with that, and than switched to the current approach:

* re-using existing headers requires some code duplication, as there are essentially two places in the code we want to trigger that. Triggering that from _repair_ also feels a bit iffy, as advancing op is not, strictly speaking, repair (the main thing we want to do here is not as much replacing headers per se, as bumping our op). Doing from checkpoint callback felt kinda random. _And_ I'd love to minimize the number of places in code where we bump the `op`, as that's super sensitive. 
* if we hit this case, we are lagging behind the cluster somewhat. So, it is plausible that something else was prepared in the view while we were checkpointing. So asking the primary to just give us the latest and greatest prepares feels like it might be more optimal, as we'll have t query those headers anyway.

Reply:
That makes perfect sense; thanks!